packages/loopring_v3/contracts/core/impl/libtransactions/TransferTransaction.sol

```diff
@@ -56,12 +62,17 @@ library TransferTransaction
     {
         // Read the transfer
         Transfer memory transfer = readTransfer(data, offset);
+        TransferAuxiliaryData memory auxData = abi.decode(auxiliaryData, (TransferAuxiliaryData));
+
+        // Check validUntil
+        require(block.timestamp < auxData.validUntil, "WITHDRAWAL_EXPIRED");
+        transfer.validUntil = auxData.validUntil;
 
         // Calculate the tx hash
         bytes32 txHash = hash(ctx.DOMAIN_SEPARATOR, transfer);
 
         // Check the on-chain authorization
-        S.requireAuthorizedTx(transfer.from, auxiliaryData, txHash);
+        S.requireAuthorizedTx(transfer.from, auxData.signature, txHash);
```

Comment:
For transfers with EdDSA signature authorization, the validUntil in the aux data will not be checked online so the operator can use any timestamp value

Reply:
It should. :) Nothing has actually been changed in the circuit, `validUntil` just isn't passed into the DA any more, so we can't count on it being correct on-chain.